dexbot/strategies/king_of_the_hill.py

```diff
@@ -0,0 +1,376 @@
+                self.log.error('Amount for {} order is too small'.format(order_type))
+                return
+            new_order = self.place_market_buy_order(amount_quote, self.buy_price)
+            self.beaten_buy_order = self.buy_order_to_beat
+        elif order_type == 'sell':
+            if not self.sell_price:
+                self.log.error('Cannot determine sell price, correct your bounds and/or ignore thresholds')
+                self.disabled = True
+                return
+            amount_quote = self.amount_quote
+            if not amount_quote:
+                self.log.error('Cannot place {} order with 0 amount. Adjust your settings!'.format(order_type))
+                self.disabled = True
+                return
+            amount_base = amount_quote * self.sell_price
+            # Prevent too small amounts
+            if self.is_too_small_amounts(amount_quote, amount_base):
+                self.log.error('Amount for {} order is too small'.format(order_type))
+                return
+            new_order = self.place_market_sell_order(amount_quote, self.sell_price)
+            self.beaten_sell_order = self.sell_order_to_beat
+
+        if new_order:
+            # Store own orders into list to perform checks later
+            self.orders.append(new_order)
+        else:
+            self.log.error('Failed to place {} order'.format(order_type))
+
+    def place_orders(self):
+        """ Place new orders
+        """
+        place_buy = False
+        place_sell = False
+
+        if self.mode == 'both':
+            place_buy = True
+            place_sell = True
+        elif self.mode == 'buy':
+            place_buy = True
+        elif self.mode == 'sell':
+            place_sell = True
+
+        if place_buy:
+            self.place_order('buy')
+        if place_sell:
+            self.place_order('sell')
+
+    @property
+    def amount_quote(self):
+        """ Get quote amount, calculate if order size is relative
+        """
+        amount = self.sell_order_amount
+        if self.is_relative_order_size:
+            quote_balance = float(self.balance(self.market['quote']))
+            amount = quote_balance * (amount / 100)
+
+        return amount
+
+    @property
+    def amount_base(self):
+        """ Get base amount, calculate if order size is relative
+        """
+        amount = self.buy_order_amount
+        if self.is_relative_order_size:
+            base_balance = float(self.balance(self.market['base']))
+            amount = base_balance * (amount / 100)
+
+        return amount
+
+    def error(self, *args, **kwargs):
+        """ Defines what happens when error occurs """
+        self.disabled = True
+
+    def tick(self, d):
+        """ Ticks come in on every block """
+        if not (self.counter or 0) % 4:
+            self.maintain_strategy()
+        self.counter += 1
+
+class StrategyController(StrategyController):
```

Comment:
Move the StrategyController into `dexbot/controllers/strategy_controller.py` and rename to `KingOfTheHillController` to stay more consistent with the rest of the strategies.

Reply:
Ok, the goal was to keep the strategy self-contained. It feels more logical to have all strategy components in single place.

Reply:
Consistency is important. If what bitfag suggests is better, then the other strategies should be changed. If that's not hard then could it be done?